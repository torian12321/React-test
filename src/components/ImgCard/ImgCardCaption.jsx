import React      from 'react';
import PropTypes  from 'prop-types';

class ImgCardCaption extends React.PureComponent {
  render() {
    return this.props.text ? (
      <figcaption>{this.props.text}</figcaption>
    ) : null
  }
}

ImgCardCaption.propTypes = {
  text: PropTypes.string
};

module.exports = ImgCardCaption;