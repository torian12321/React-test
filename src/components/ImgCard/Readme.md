Styled version of [figure](https://www.w3schools.com/tags/tag_figure.asp)

#### Basic
Use this just for wireframing
```
<ImgCard />
```

#### With alt and caption text
```
<ImgCard alt='Alt text' caption='Caption text' />
```

#### With alt text
Caption will be equal to caption text
```
<ImgCard alt='Alt text' />
```

#### Custom src
Use a local image or from a url
```
const imgExample = require ('../Img/example.jpg');

<div>
  <ImgCard alt="img 1" src='http://placeimg.com/200/150/nature' />
  <ImgCard alt="img 2" src={imgExample} />
</div>
```
