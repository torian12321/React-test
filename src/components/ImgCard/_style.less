@import '../../styles/conf/index.less';

@ImgCard-captionBG: @black;

.imageCard{
    position: relative;
    overflow: hidden;
    .rounded();
    
    img{
        width  : 100%;
        display: block;
    }
    figcaption{
        position  : absolute;
        bottom    : 0;
        left      : 0;
        right     : 0;
        padding   : .4em 1em;
        background: @ImgCard-captionBG;
        opacity   : 0;
        .transition(opacity);
        .smart-text-color(@ImgCard-captionBG);
    }
    &:hover{
        figcaption{
            opacity: .6;
        }
    }
}