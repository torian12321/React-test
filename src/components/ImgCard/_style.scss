@import '../../styles/conf/index.scss';

// Vars
$ImgCard-captionBG: $black !default;

// Block
.imageCard{
  position: relative;
  overflow: hidden;
  @include rounded();

  img{
    width  : 100%;
    display: block;
  }
  figcaption{
    position  : absolute;
    bottom    : 0;
    left      : 0;
    right     : 0;
    padding   : .4em 1em;
    background: $ImgCard-captionBG;
    opacity   : 0;
    @include transition(opacity);
    @include smart-text-color($ImgCard-captionBG);
  }
  &:hover{
    figcaption{
      opacity: .6;
    }
  }
}