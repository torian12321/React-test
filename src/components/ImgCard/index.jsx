import React      from 'react';
import PropTypes  from 'prop-types';
import classNames from 'classnames';
import Img        from '../Img';
import './_style.scss';

class ImgCard extends React.PureComponent {
	render() {
    const captionString = this.props.caption || this.props.alt || null;

		return(
      <figure className={classNames('imageCard', this.props.className)}>
        <Img className='imageCard__img' src={this.props.src} alt={this.props.alt} />
        { captionString &&
          <figcaption className='imageCard__caption'>{captionString}</figcaption>
        }
      </figure>
		);
	}
}

ImgCard.propTypes = {
  className: PropTypes.string,
  src      : PropTypes.string.isRequired,
  alt      : PropTypes.string,
  caption  : PropTypes.string
};

module.exports = ImgCard;