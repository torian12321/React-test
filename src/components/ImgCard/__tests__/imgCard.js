import React    from 'react';
import renderer from 'react-test-renderer';
import ImgCard  from '../index';

describe('component', () => {
  describe('<ImgCard />', () => {

    it('should render correctly', () => {
      expect(renderer.create(
        <ImgCard src='img.png' />
      )).toMatchSnapshot();
    });

    it('should render with alt text', () => {
      expect(renderer.create(
        <ImgCard src='img.png' alt='image description' />
      )).toMatchSnapshot();
    });

    it('should render with alt text', () => {
      expect(renderer.create(
        <ImgCard src='img.png' alt='image description' caption='caption text' />
      )).toMatchSnapshot();
    });

  });
});