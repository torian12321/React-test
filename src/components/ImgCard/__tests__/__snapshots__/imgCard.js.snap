// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component <ImgCard /> should render correctly 1`] = `
<figure
  className="imageCard"
>
  <img
    alt={undefined}
    role="presentation"
    src="img.png"
  />
</figure>
`;

exports[`component <ImgCard /> should render with alt text 1`] = `
<figure
  className="imageCard"
>
  <img
    alt="image description"
    role={null}
    src="img.png"
  />
  <figcaption>
    image description
  </figcaption>
</figure>
`;

exports[`component <ImgCard /> should render with alt text 2`] = `
<figure
  className="imageCard"
>
  <img
    alt="image description"
    role={null}
    src="img.png"
  />
  <figcaption>
    caption text
  </figcaption>
</figure>
`;
